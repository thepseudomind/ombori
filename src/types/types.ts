export interface IMain{
    status: boolean,
    toggleSidebar?: any
}

export interface IProduct{
    id: number,
    email: string,
    first_name: string,
    last_name: string,
    avatar: string
}

export type TLoader = {
    mini: boolean
}

export type TMessage = {
    message: string
}