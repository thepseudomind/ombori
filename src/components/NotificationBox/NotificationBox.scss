.notification{
    position: fixed;
    bottom: 1rem;
    left: 2rem;
    height: 4rem;
    width: 15rem;
    border-radius: .5rem;
    background-color: white;
    box-shadow: 0 1rem 2rem rgba(0, 0, 0, 0.3);
    opacity: 0;
    visibility: hidden;
    animation: fadeInOut 3s none ease;
    display: flex;
    justify-content: center;
    align-items: center;
    z-index: 2;

    &__message{
        font-size: 1.25rem;
        font-weight: 600;
        text-transform: uppercase;
    }
}

@keyframes fadeInOut{
    0%{
        transform: translateY(-2rem);
        opacity: 0;
        visibility: hidden;
    }

    50%{
        transform: translateY(0);
        opacity: 1;
        visibility: visible;
    }

    100%{
        transform: translateY(0);
        opacity: .5;
        visibility: hidden;
    }
}