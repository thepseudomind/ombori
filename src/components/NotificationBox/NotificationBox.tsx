import React from 'react';
import './NotificationBox.scss';
import { TMessage } from '../../types/types';

const NotificationBox = ({message} : TMessage)=>{
    return (
        <div className="notification">
            <p className="notification__message">No more users</p>
        </div>
    );
}

export default NotificationBox;