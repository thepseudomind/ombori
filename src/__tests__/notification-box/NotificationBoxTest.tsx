import React from 'react';
import { shallow } from  'enzyme';
import NotificationBox from '../../components/NotificationBox/NotificationBox';
import { TMessage } from '../../types/types';

describe('notification test', ()=>{
    const mockProps : TMessage = {
        message: ''
    };
    const wrapper = shallow(<NotificationBox {...mockProps}/>);

    it('shallow render', ()=>{
        expect(wrapper).toMatchSnapshot();
    });
});
