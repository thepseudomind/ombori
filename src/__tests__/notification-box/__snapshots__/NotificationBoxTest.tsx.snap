// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`notification test shallow render 1`] = `
<div
  className="notification"
>
  <p
    className="notification__message"
  >
    No more users
  </p>
</div>
`;
